packages/near-membrane-base/src/membrane.ts

```diff
@@ -1327,34 +1368,32 @@ export function createMembraneMarshall(
             // The closure works as the implicit WeakMap.
             const targetPointer = createPointer(distortedTarget);
             const targetTraits = getTargetTraits(distortedTarget);
+            let targetFunctionArity: number | undefined;
             let targetFunctionName: string | undefined;
-            if (typeof originalTarget === 'function') {
-                if (DEV_MODE) {
-                    try {
-                        // A revoked proxy will throw when reading the function name.
-                        const unsafeNameDesc = ReflectGetOwnPropertyDescriptor(
-                            originalTarget,
-                            'name'
-                        );
-                        if (unsafeNameDesc) {
-                            const safeNameDesc = toSafeDescriptor(unsafeNameDesc);
-                            const { value: safeNameDescValue } = safeNameDesc;
-                            if (typeof safeNameDescValue === 'string') {
-                                targetFunctionName = safeNameDescValue;
-                            }
-                        }
-                    } catch {
-                        // Intentionally swallowing the error because this method
-                        // is just extracting the function in a way that it should
-                        // always succeed except for the cases in which the provider
-                        // is a proxy that is either revoked or has some logic to
-                        // prevent reading the name property descriptor.
+            if (targetTraits & TargetTraits.IsFunction) {
+                const unsafeLengthDesc = ReflectGetOwnPropertyDescriptor(originalTarget, 'length');
+                if (unsafeLengthDesc) {
+                    const safeLengthDesc = toSafeDescriptor(unsafeLengthDesc);
+                    const { value: safeLengthDescValue } = safeLengthDesc;
+                    if (typeof safeLengthDescValue === 'number') {
+                        targetFunctionArity = safeLengthDescValue;
+                    }
+                }
```

Comment:
☝️ Tracking arity is an extra lookup set of actions hopefully mitigated by all the inlining elsewhere.